### PR TITLE
flavours: 0.4.0 -> 0.5.0

### PR DESCRIPTION
**Pull request by Misterio77**
###### Motivation for this change
Update to follow upstream, add missing shell (bash, zsh, fish) completions.
###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Obs: Although i'm the upstream developer, i just started using NixOS this week (loving it). I did read about packaging, but it's possible i missed something. Please let me know if there's anything to be fixed.

### CODE REVIEW

File: pkgs/applications/misc/flavours/default.nix [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
     rev = "v${version}";
-    sha256 = "sha256-rDy859jg+F8XC4sJogIgdn1FoT8cf7S+KORt+7kboAc=";
+    sha256 = "1bgi6p7l0bh9k4vkwvngk7q19ynia0z1ninb1cq8qnwwpll6kbya";
   };
 
   buildInputs = [ ]
     ++ lib.optionals stdenv.isDarwin [ libiconv ];
 
-  cargoSha256 = "sha256-I8ZH35L2CVLy6ypmdOPd8VEG/sQeGaHyT1HWNdwyZVo=";
+  cargoSha256 = "sha256:07hwxhfcbqbwb3hz18w92h1lhdiwwy7abhwpimzx7syyavp4rmn4";
```
Comment (fortuneteller2k) [inline pkgs/applications/misc/flavours/default.nix:17]:
```suggestion
  cargoSha256 = "07hwxhfcbqbwb3hz18w92h1lhdiwwy7abhwpimzx7syyavp4rmn4";
```

Comment (fortuneteller2k) [reply pkgs/applications/misc/flavours/default.nix:17]: otherwise, LGTM :)

Comment (Misterio77) [reply pkgs/applications/misc/flavours/default.nix]: Thanks for the suggestion!

**Review by moni-dz — APPROVED** [review]

**Review by moni-dz — APPROVED** [review]

File: pkgs/applications/misc/flavours/default.nix [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
-  cargoSha256 = "sha256-I8ZH35L2CVLy6ypmdOPd8VEG/sQeGaHyT1HWNdwyZVo=";
+  cargoSha256 = "07hwxhfcbqbwb3hz18w92h1lhdiwwy7abhwpimzx7syyavp4rmn4";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    $out/bin/flavours --completions bash > flavours.bash
+    $out/bin/flavours --completions fish > flavours.fish
+    $out/bin/flavours --completions zsh > _flavours
+    installShellCompletion --zsh _flavours --fish flavours.fish --bash flavours.bash
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/flavours/default.nix:25]:
```suggestion
    installShellCompletion --cmd flavours \
      --zsh <($out/bin/flavours --completions zsh) \
      --fish <($out/bin/flavours --completions fish) \
      --bash <($out/bin/flavours --completions bash)
```

Comment (Misterio77) [reply pkgs/applications/misc/flavours/default.nix:25]: Very elegant use of process substitution. Thanks for the suggestion!

File: pkgs/applications/misc/flavours/default.nix [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
   src = fetchFromGitHub {
     owner = "Misterio77";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-rDy859jg+F8XC4sJogIgdn1FoT8cf7S+KORt+7kboAc=";
+    sha256 = "1bgi6p7l0bh9k4vkwvngk7q19ynia0z1ninb1cq8qnwwpll6kbya";
   };
 
   buildInputs = [ ]
     ++ lib.optionals stdenv.isDarwin [ libiconv ];
```
[the remark this answers is not in the document]
Comment (Misterio77) [reply pkgs/applications/misc/flavours/default.nix:15]: Thanks, this does look more elegant without list concatenation.

**Review by moni-dz — APPROVED** [review]

**Review by moni-dz — APPROVED** [review]

**Review by jonringer — APPROVED** [review]
LGTM

```
https://github.com/NixOS/nixpkgs/pull/128558

1 package built:
flavours
```

**Comment by jonringer** [comment]
thanks @Misterio77 for your first nixpkgs contribution :tada: :)

**Pull request merged**